pontoon/base/static/js/themeSwitcher.js

```diff
@@ -0,0 +1,77 @@
+$(function () {
+  function getSystemTheme() {
+    if (
+      window.matchMedia &&
+      window.matchMedia('(prefers-color-scheme: dark)').matches
+    ) {
+      return 'dark';
+    } else {
+      return 'light';
+    }
+  }
+
+  function updateTheme(newTheme) {
+    if (newTheme === 'system') {
+      newTheme = getSystemTheme();
+    }
+    $('body')
+      .removeClass('dark-theme light-theme system-theme')
+      .addClass(`${newTheme}-theme`);
+
+    if (newTheme !== 'system') {
+      Pontoon.endLoader(`Theme changed to ${newTheme}.`);
+    }
+  }
+
+  window
+    .matchMedia('(prefers-color-scheme: dark)')
+    .addEventListener('change', function (e) {
+      if ($('.appearance .toggle-button button.system').hasClass('active')) {
+        if (e.matches) {
+          updateTheme('dark');
+        } else {
+          updateTheme('light');
+        }
+      }
+    });
+
+  $(document).ready(function () {
+    if ($('body').hasClass('system-theme')) {
+      let systemTheme = getSystemTheme();
+      $('body').removeClass('system-theme').addClass(`${systemTheme}-theme`);
+    }
+  });
+
+  $('.appearance .toggle-button button').click(function (e) {
+    e.preventDefault();
+
+    var self = $(this);
+
+    if (self.is('.active')) {
+      return;
+    }
+
+    var theme = self.attr('class').split(' ')[0];
```

Comment:
This looks fragile. I suggest we also put values in button `value` attributes and then just use `self.val()` here.